atomic_reactor/plugins/post_group_manifests.py

```diff
@@ -0,0 +1,112 @@
+"""
+
+
+from __future__ import unicode_literals
+from copy import deepcopy
+import requests
+import requests.auth
+import json
+
+try:
+    from urlparse import urlparse
+except ImportError:
+    from urllib.parse import urlparse
+
+from atomic_reactor.plugin import PostBuildPlugin
+from atomic_reactor.plugins.build_orchestrate_build import OrchestrateBuildPlugin
+from atomic_reactor.util import Dockercfg
+
+
+class GroupManifestsPlugin(PostBuildPlugin):
+    key = 'group_manifests'
+    is_allowed_to_fail = False
+
+    def __init__(self, tasker, workflow, registries, group=True, goarch={}):
+        """
+        constructor
+
+        :param tasker: DockerTasker instance
+        :param workflow: DockerBuildWorkflow instance
+        :param timeout: int, maximum number of seconds to wait
+        :param retry_delay: int, seconds between pull attempts
+        """
+        # call parent constructor
+        super(GroupManifestsPlugin, self).__init__(tasker, workflow)
+        self.group = group
+        self.goarch = goarch
+        self.registries = deepcopy(registries)
+
+    def get_worker_manifest(self, worker_data):
+        worker_digests = worker_data['digests']
+        worker_manifest = []
+
+        for registry, registry_conf in self.registries.items():
+            if not registry.startswith('http://') and not registry.startswith('https://'):
+                registry = 'https://' + registry
+
+            registry_noschema = urlparse(registry).netloc
+            for push_conf_registry in self.workflow.push_conf.docker_registries:
+                if push_conf_registry.uri == registry_noschema:
+                    break
+            else:
+                continue
+
+            insecure = push_conf_registry.insecure
+            auth = None
+            secret_path = registry_conf.get('secret')
+            if secret_path:
+                self.log.debug("registry %s secret %s", registry_noschema, secret_path)
+                dockercfg = Dockercfg(secret_path).get_credentials(registry_noschema)
+                try:
+                    username = dockercfg['username']
+                    password = dockercfg['password']
+                except KeyError:
+                    self.log.error("credentials for registry %s not found in %s",
+                                   registry_noschema, secret_path)
+                else:
+                    self.log.debug("found user %s for registry %s", username, registry_noschema)
+                    auth = requests.auth.HTTPBasicAuth(username, password)
+
+            digest = worker_digests[0]['digest']
+            tag = worker_digests[0]['tag']
+            repo = tag.split(':')[0]
+            for image in self.workflow.tag_conf.images:
+                image_tag = image.to_str(registry=False)
+
+                url = '{}/v2/{}/manifests/{}'.format(registry, repo, digest)
+
+                response = requests.head(url, verify=not insecure, auth=auth)
```

Comment:
`HEAD` should be sufficient for checking that the manifest exists, but it also should have `response.raise_for_status()` to throw exception on HTTP 404.

Anyway, since `get` is performed later on, its not needed